fairworkflows/fairstep.py

```diff
@@ -19,26 +22,21 @@ class FairStep(RdfWrapper):
 
     DEFAULT_STEP_URI = 'http://purl.org/nanopub/temp/mynanopub#step'
 
-    def __init__(self, step_rdf: rdflib.Graph = None, uri=DEFAULT_STEP_URI,
-                 from_nanopub=False, func=None):
+    def __init__(self, uri=DEFAULT_STEP_URI):
         super().__init__(uri=uri)
 
-        if func:
-            self.from_function(func)
-        elif from_nanopub:
-            self.load_from_nanopub(uri)
-        else:
-            self._uri = uri
-
-            if step_rdf:
-                self._rdf = step_rdf
-
-                if self._uri not in step_rdf.subjects():
-                    print(f"Warning: Provided URI '{self._uri}' does not match any subject in provided rdf graph.")
-            else:
-                self._rdf = rdflib.Graph()
-
-    def load_from_nanopub(self, uri):
+    @classmethod
+    def from_rdf(cls, rdf):
```

Comment:
I didn't realise this existed, but now that I refactored this in this explicit form, this basically solves #72. So I guess only thing remaining there is to also add a similar method to FairWorkFlow.

Reply:
I agree - this is a much cleaner looking way of creating the Fair elements from the rdf. I have never needed to use ```@classmethod``` before, but it works nicely here. Great job refactoring! I see you have now also added a similar method to ```FairWorkflow``` in #76 so that's perfect.